config/v1/types_authentication.go

```diff
@@ -253,9 +254,16 @@ type TokenIssuer struct {
 	// The Kubernetes API server determines how authentication tokens should be handled
 	// by matching the 'iss' claim in the JWT to the issuerURL of configured identity providers.
 	//
-	// issuerURL must use the 'https' scheme.
+	// Must be at least 1 character and must not exceed 267 characters in length.
+	// Must be a valid URL that uses the 'https' scheme and does not contain a query, fragment or user.
 	//
-	// +kubebuilder:validation:Pattern=`^https:\/\/[^\s]`
+	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.orValue(\"\") == self) || isURL(self)",message="must be a valid URL",optionalOldSelf=true
+	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.orValue(\"\") == self) || url(self).getScheme() == 'https'",message="must use the 'https' scheme",optionalOldSelf=true
+	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.orValue(\"\") == self) || url(self).getQuery() == {}",message="must not have a query",optionalOldSelf=true
+	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.orValue(\"\") == self) || self.find('#(.+)$') == ''",message="must not have a fragment",optionalOldSelf=true
+	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.orValue(\"\") == self) || self.find('@') == ''",message="must not have user info",optionalOldSelf=true
```

Comment:
These can be simplified
```suggestion
	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.value() == self) || isURL(self)",message="must be a valid URL",optionalOldSelf=true
	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.value() == self) || url(self).getScheme() == 'https'",message="must use the 'https' scheme",optionalOldSelf=true
	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.value() == self) || url(self).getQuery() == {}",message="must not have a query",optionalOldSelf=true
	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.value() == self) || self.find('#(.+)$') == ''",message="must not have a fragment",optionalOldSelf=true
	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.value() == self) || self.find('@') == ''",message="must not have user info",optionalOldSelf=true
```